### ISSUE TITLE
For hap training, it stopped after step 47999

### ISSUE THREAD

**Issue by zhangtao22** [Wangt-CN/DisCo#13]
my log is like this 
![1689042442307](https://github.com/Wangt-CN/DisCo/assets/35836562/95abc04b-e464-43fd-a8cd-9c88587e9cd6)

as for metric.json, it is 
{"Step 0": {"eval": {"FID": 290.6777284435151, "time": "0:03:31.655149"}}, "Epoch2": {"train": {"loss_total": 0.09805237877084273, "time": "2:00:58.214095"}}, "Step2000": {"eval": {"FID": 39.84386908484396, "time": "0:04:10.178784"}}, "Epoch3": {"train": {"loss_total": 0.08273238215732012, "time": "1:05:12.567647"}}, "Step4000": {"eval": {"FID": 43.90990567108605, "time": "0:04:07.709888"}}, "Epoch4": {"train": {"loss_total": 0.08007360270170316, "time": "0:12:27.958903"}}, "Epoch5": {"train": {"loss_total": 0.07951762963632482, "time": "2:10:26.386287"}}, "Step6000": {"eval": {"FID": 47.38045255087343, "time": "0:04:09.181442"}}, "Epoch6": {"train": {"loss_total": 0.0778031051322654, "time": "1:17:38.705795"}}, "Step8000": {"eval": {"FID": 44.090458160650826, "time": "0:04:09.042723"}}, "Epoch7": {"train": {"loss_total": 0.07623744776395902, "time": "0:24:55.471577"}}, "Epoch8": {"train": {"loss_total": 0.07622077868163159, "time": "2:22:49.398254"}}, "Step10000": {"eval": {"FID": 31.904819727152358, "time": "0:04:09.004639"}}, "Epoch9": {"train": {"loss_total": 0.07504117791188147, "time": "1:30:05.510775"}}, "Step12000": {"eval": {"FID": 27.483082697985367, "time": "0:04:09.374085"}}, "Epoch10": {"train": {"loss_total": 0.07409243798521284, "time": "0:37:20.796301"}}, "Epoch11": {"train": {"loss_total": 0.07395339482515068, "time": "2:35:10.650475"}}, "Step14000": {"eval": {"FID": 31.168737757947156, "time": "0:04:09.885283"}}, "Epoch12": {"train": {"loss_total": 0.07372492651599219, "time": "1:42:29.514715"}}, "Step16000": {"eval": {"FID": 27.21106589500107, "time": "0:04:08.266375"}}, "Epoch13": {"train": {"loss_total": 0.07312383905231748, "time": "0:49:47.800272"}}, "Epoch14": {"train": {"loss_total": 0.07289745142666333, "time": "2:47:37.351242"}}, "Step18000": {"eval": {"FID": 23.106254980103927, "time": "0:04:07.923059"}}, "Epoch15": {"train": {"loss_total": 0.07242165016734459, "time": "1:54:56.220633"}}, "Step20000": {"eval": {"FID": 27.248582831371834, "time": "0:04:08.544414"}}, "Epoch16": {"train": {"loss_total": 0.07194297805632631, "time": "1:02:14.190223"}}, "Step22000": {"eval": {"FID": 24.803106175247194, "time": "0:04:07.721933"}}, "Epoch17": {"train": {"loss_total": 0.07178588963246771, "time": "0:09:32.512436"}}, "Epoch18": {"train": {"loss_total": 0.07133925958989136, "time": "2:07:23.062100"}}, "Step24000": {"eval": {"FID": 24.043788111684535, "time": "0:04:08.888000"}}, "Epoch19": {"train": {"loss_total": 0.07101746586461861, "time": "1:14:44.514627"}}, "Step26000": {"eval": {"FID": 21.995370168790316, "time": "0:04:09.391666"}}, "Epoch20": {"train": {"loss_total": 0.0711026608135349, "time": "0:21:59.766503"}}, "Epoch21": {"train": {"loss_total": 0.07048133608498951, "time": "2:19:49.997902"}}, "Step28000": {"eval": {"FID": 24.72485237611329, "time": "0:04:08.996139"}}, "Epoch22": {"train": {"loss_total": 0.07035766966611788, "time": "1:27:07.072584"}}, "Step30000": {"eval": {"FID": 23.718398035524274, "time": "0:04:08.026119"}}, "Epoch23": {"train": {"loss_total": 0.07033483951472409, "time": "0:34:28.609709"}}, "Epoch24": {"train": {"loss_total": 0.0700808005451255, "time": "2:32:20.204781"}}, "Step32000": {"eval": {"FID": 22.28186004474327, "time": "0:04:09.402551"}}, "Epoch25": {"train": {"loss_total": 0.06941193997643072, "time": "1:39:37.250015"}}, "Step34000": {"eval": {"FID": 22.008737701972393, "time": "0:04:08.751428"}}, "Epoch26": {"train": {"loss_total": 0.06975648029961369, "time": "0:46:56.325861"}}, "Epoch27": {"train": {"loss_total": 0.06926694908420368, "time": "2:44:55.461831"}}, "Step36000": {"eval": {"FID": 20.646719371436802, "time": "0:04:09.315956"}}, "Epoch28": {"train": {"loss_total": 0.06902978069161715, "time": "1:51:58.898925"}}, "Step38000": {"eval": {"FID": 20.947136795766653, "time": "0:04:08.238444"}}, "Epoch29": {"train": {"loss_total": 0.06883154165577786, "time": "0:59:21.183679"}}, "Step40000": {"eval": {"FID": 21.91535396280665, "time": "0:04:08.135138"}}, "Epoch30": {"train": {"loss_total": 0.0685038694108908, "time": "0:06:39.512788"}}, "Epoch31": {"train": {"loss_total": 0.06833649117448559, "time": "2:04:39.496344"}}, "Step42000": {"eval": {"FID": 21.03997708430046, "time": "0:04:08.954742"}}, "Epoch32": {"train": {"loss_total": 0.06803931710874384, "time": "1:11:49.735317"}}, "Step44000": {"eval": {"FID": 20.869328025712207, "time": "0:04:08.273694"}}, "Epoch33": {"train": {"loss_total": 0.06827863852959126, "time": "0:19:08.113691"}}, "Epoch34": {"train": {"loss_total": 0.06779086924764614, "time": "2:16:59.691355"}}, "Step46000": {"eval": {"FID": 21.228485860542378, "time": "0:04:07.829478"}}, "Epoch35": {"train": {"loss_total": 0.06772437718459348, "time": "1:24:18.001555"}}, "Step48000": {"eval": {"FID": 21.43211396473822, "time": "0:04:08.974074"}}, "Epoch36": {"train": {"loss_total": 0.06746692015109836, "time": "0:31:33.721808"}}}
Does the code have the mechanism of early stop or my code have encounter sth error?
I ran this code using this
AZFUSE_USE_FUSE=0 QD_USE_LINEIDX_8B=0 NCCL_ASYNC_ERROR_HANDLING=0 mpirun -np 8 --allow-run-as-root python finetune_sdm_yaml.py --cf config/ref_attn_clip_combine_controlnet_attr_pretraining/coco_S256_xformers_tsv_strongrand.py --do_train --root_dir /data/DisCo --local_train_batch_size 64 --local_eval_batch_size 64 --log_dir exp/tiktok_pretrain --epochs 40 --deepspeed --eval_step 2000 --save_step 2000 --gradient_accumulate_steps 1 --learning_rate 1e-3 --fix_dist_seed --loss_target "noise" --train_yaml ./TSV_dataset/Human_Attribute_Pretrain/composite/train_TiktokDance-coco-single_person-Lindsey_0411_youtube-SHHQ-1.0-deepfashion2-laion_human-masks-single_cap.yaml --val_yaml ./TSV_dataset/Human_Attribute_Pretrain/composite/val_TiktokDance-coco-single_person-SHHQ-1.0-masks-single_cap.yaml --unet_unfreeze_type "transblocks" --refer_sdvae --ref_null_caption False --combine_clip_local --combine_use_mask --conds "masks" --max_eval_samples 2000 --strong_aug_stage1 --node_split_sampler 0 >> log.txt 2>&1

**Comment by zhangtao22** [comment]
def setup_model_for_training(self):
        if self.args.resume:
            if not self.args.deepspeed:
                if os.path.exists(os.path.join(self.log_dir, self.last_checkpoint_filename)):
                    logger.warning(f'Resuming.... from {self.last_checkpoint_filename}')
                    self.load_checkpoint(self.last_checkpoint_filename)
            else:        
                if isinstance(self.optimizer, list):
                    raise ValueError("self.optimizer is a list, which is incompatible with deepspeed")
if I wanna resume，I should not write this --deepspeed in the command line？

**Comment by Wangt-CN** [comment]
Hi @zhangtao22, actually I do not meet the early stop before. For resuming, you may not deprecate the `--deepspeed`. Could you please try to add `--pretrained_model {path to your early stop model, xxx.pt file}` into the command line?

**Comment by zhangtao22** [comment]
@Wangt-CN if I finetune_sdm_yaml.py --pretrained_model {/data/DisCo/exp/tiktok_pretrain/35999.pth, mp_rank_00_model_states.pt},it showed finetune_sdm_yaml.py: error: unrecognized arguments: mp_rank_00_model_states.pt}

**Comment by zhangtao22** [comment]
if AZFUSE_USE_FUSE=0 QD_USE_LINEIDX_8B=0 NCCL_ASYNC_ERROR_HANDLING=0 mpirun -np 2 --allow-run-as-root python finetune_sdm_yaml.py --pretrained_model /data/DisCo/exp/tiktok_pretrain/35999.pth/mp_rank_00_model_states.pt --cf config/ref_attn_clip_combine_controlnet_attr_pretraining/coco_S256_xformers_tsv_strongrand.py --do_train --root_dir /data/DisCo --local_train_batch_size 64 --local_eval_batch_size 64 --log_dir exp/tiktok_pretrain --epochs 40 --resume --eval_step 2000 --save_step 2000 --gradient_accumulate_steps 1 --learning_rate 1e-3 --fix_dist_seed --loss_target "noise" --train_yaml ./TSV_dataset/Human_Attribute_Pretrain/composite/train_TiktokDance-coco-single_person-Lindsey_0411_youtube-SHHQ-1.0-deepfashion2-laion_human-masks-single_cap.yaml --val_yaml ./TSV_dataset/Human_Attribute_Pretrain/composite/val_TiktokDance-coco-single_person-SHHQ-1.0-masks-single_cap.yaml --unet_unfreeze_type "transblocks" --refer_sdvae --ref_null_caption False --combine_clip_local --combine_use_mask --conds "masks" --max_eval_samples 2000 --strong_aug_stage1 --node_split_sampler 0
the log is this.
[2023-07-17 16:00:46 <wutils_ldm.py:457> file2data] Loaded data from /data/DisCo/exp/tiktok_pretrain/35999.pth/mp_rank_00_model_states.pt^[[0m
dict_keys(['module', 'buffer_names', 'optimizer', 'param_shapes', 'lr_scheduler', 'sparse_tensor_module_names', 'skipped_steps', 'global_steps', 'global_samples', 'dp_world_size', 'mp_world_size', 'ds_config', 'ds_version'])
Optimizer State Dict:
^[[33m[2023-07-17 16:00:47 <wutils_ldm.py:690> adaptively_load_state_dict] Some weights of state_dict were not used in target: ['module', 'buffer_names', 'optimizer', 'param_shapes', 'lr_scheduler', 'sparse_tensor_module_names', 'skipped_steps', 'global_steps', 'global_samples', 'dp_world_size', 'mp_world_size', 'ds_config', 'ds_version']^[[0m
^[[33m[2023-07-17 16:00:47 <wutils_ldm.py:690> adaptively_load_state_dict] Some weights of state_dict were not used in target: ['module', 'buffer_names', 'optimizer', 'param_shapes', 'lr_scheduler', 'sparse_tensor_module_names', 'skipped_steps', 'global_steps', 'global_samples', 'dp_world_size', 'mp_world_size', 'ds_config', 'ds_version']^[[0m
^[[33m[2023-07-17 16:00:47 <wutils_ldm.py:698> adaptively_load_state_dict] Some weights of target are missing in state_dict: ['vae.encoder.conv_in.weight', 'vae.encoder.conv_in.bias', 'vae.encoder.down_blocks.0.resnets.0.norm1.weight', 'vae.encoder.down_blocks.0.resnets.0.norm1.bias', 'vae.encoder.down_blocks.0.resnets.0.conv1.weight', 'vae.encoder.down_blocks.0.resnets.0.conv1.bias', 'vae.encoder.down_blocks.0.resnets.0.norm2.weight', 'vae.encoder.down_blocks.0.resnets.0.norm2.bias', 'vae.encoder.down_blocks.0.resnets.0.conv2.weight', 'vae.encoder.down_blocks.0.resnets.0.conv2.bias', 'vae.encoder.down_blocks.0.resnets.1.norm1.weight', 'vae.encoder.down_blocks.0.resnets.1.norm1.bias', 'vae.encoder.down_blocks.0.resnets.1.conv1.weight', 'vae.encoder.down_blocks.0.resnets.1.conv1.bias', 'vae.encoder.down_blocks.0.resnets.1.norm2.weight', 'vae.encoder.down_blocks.0.resnets.1.norm2.bias', 'vae.encoder.down_blocks.0.resnets.1.conv2.weight', 'vae.encoder.down_blocks.0.resnets.1.conv2.bias', 'vae.encoder.down_blocks.0.downsamplers.0.conv.weight', 'vae.encoder.down_blocks.0.downsamplers.0.conv.bias', 'vae.encoder.down_blocks.1.resnets.0.norm1.weight', 'vae.encoder.down_blocks.1.resnets.0.norm1.bias', 'vae.encoder.down_blocks.1.resnets.0.conv1.weight', 'vae.encoder.down_blocks.1.resnets.0.conv1.bias', 'vae.encoder.down_blocks.1.resnets.0.norm2.weight', 'vae.encoder.down_blocks.1.resnets.0.norm2.bias', 'vae.encoder.down_blocks.1.resnets.0.conv2.weight', 'vae.encoder.down_blocks.1.resnets.0.conv2.bias', 'vae.encoder.down_blocks.1.resnets.0.conv_shortcut.weight', 'vae.encoder.down_blocks.1.resnets.0.conv_shortcut.bias', 'vae.encoder.down_blocks.1.resnets.1.norm1.weight', 'vae.encoder.down_blocks.1.resnets.1.norm1.bias', 'vae.encoder.down_blocks.1.resnets.1.conv1.weight', 'vae.encoder.down_blocks.1.resnets.1.conv1.bias', 'vae.encoder.down_blocks.1.resnets.1.norm2.weight', 'vae.encoder.down_blocks.1.resnets.1.norm2.bias', 'vae.encoder.down_blocks.1.resnets.1.conv2.weight', 'vae.encoder.down_blocks.1.resnets.1.conv2.bias', 'vae.encoder.down_blocks.1.downsamplers.0.conv.weight', 'vae.encoder.down_blocks.1.downsamplers.0.conv.bias', 'vae.encoder.down_blocks.2.resnets.0.norm1.weight', 'vae.encoder.down_blocks.2.resnets.0.norm1.bias', 'vae.encoder.down_blocks.2.resnets.0.conv1.weight', 'vae.encoder.down_blocks.2.resnets.0.conv1.bias', 'vae.encoder.down_blocks.2.resnets.0.norm2.weight', 'vae.encoder.down_blocks.2.resnets.0.norm2.bias', 'vae.encoder.down_blocks.2.resnets.0.conv2.weight', 'vae.encoder.down_blocks.2.resnets.0.conv2.bias', 'vae.encoder.down_blocks.2.resnets.0.conv_shortcut.weight', 'vae.encoder.down_blocks.2.resnets.0.conv_shortcut.bias', 'vae.encoder.down_blocks.2.resnets.1.norm1.weight', 'vae.encoder.down_blocks.2.resnets.1.norm1.bias', 'vae.encoder.down_blocks.2.resnets.1.conv1.weight', 'vae.encoder.down_blocks.2.resnets.1.conv1.bias', 'vae.encoder.down_blocks.2.resnets.1.norm2.weight', 'vae.encoder.down_blocks.2.resnets.1.norm2.bias',

def from_pretrained(self, pretrained_model):
        if hasattr(self.model, "module"):
            raise ValueError("Please do not load pretrained models into wrapped models, ensure self.models is CPU.")
        if isinstance(pretrained_model, str):
            logger.warning('Loading Pretrained Model Path: %s...' % pretrained_model)
            pretrained_dict = file2data(pretrained_model, map_location='cpu')
            print(pretrained_dict.keys())
            print("Optimizer State Dict:")

            #print("Global Step:", pretrained_dict['global_step'])
            #print("Epoch:", pretrained_dict['epoch'])

            if 'models' in pretrained_dict:
                pretrained_dict = pretrained_dict['models']
            elif 'model' in pretrained_dict:
                pretrained_dict = pretrained_dict['model']
        else:
            logger.warning('Loading Given Pretrained Dict...')
            pretrained_dict = pretrained_model
        adaptively_load_state_dict(self.model, pretrained_dict)
I don't think that it can work
